### ISSUE TITLE
linter: `--type-aware` reports 0 findings and says nothing when a file belongs to no tsconfig program

### ISSUE THREAD

**Issue by oddharsh** [oxc-project/oxc#26822]
### What version of Oxlint are you using?

1.83.0, with oxlint-tsgolint 7.0.2001

### What command did you run?

```
oxlint --type-aware src
```

### What does your `.oxlintrc.json` config file look like?

```json
{ "rules": { "typescript/no-floating-promises": "error" } }
```

### What happened?

When a linted `.ts` file is not included by any discoverable `tsconfig.json`, tsgolint puts it in an empty default program where every name it cannot resolve is an error type, and every type-aware rule passes. oxlint prints nothing: no warning, no "0 files typed", exit 0. The only place the fact appears is `OXC_LOG=debug`, where tsgolint says it outright:

```
Got tsconfig for file <repo>/src/a.ts: <none>
Done assigning files to programs. Total programs: 0. Unmatched files: 1
  Unmatched file: <repo>/src/a.ts
Workload distribution: 0 programs
```

Minimal repro, three files:

```ts
// types/globals.d.ts
declare function fetchThing(): Promise<void>;
```
```ts
// src/a.ts
export function run() {
  fetchThing();
}
```
```json
// .oxlintrc.json
{ "rules": { "typescript/no-floating-promises": "error" } }
```

| tsconfig state | findings | any message |
|---|---|---|
| no `tsconfig.json` anywhere | 0 | none, exit 0 |
| `tsconfig.json` present, `include: ["types"]` (does not cover `src/`) | 0 | none, exit 0 |
| `tsconfig.json` with `include: ["src", "types"]` | 1 (`no-floating-promises` at `src/a.ts:2:3`) | |
| solution-style root (`files: []`, one `references` entry) covering `src` | 1 | |

By contrast, `--tsconfig nope.json` on a file that does not exist fails loudly:

```
The tsconfig file ".../nope.json" does not exist, Please provide a valid tsconfig file.
```

So an explicit path that resolves to nothing is an error, and auto-discovery that resolves to nothing is a clean pass. The second is the default and the dangerous one.

How we hit it: a repository whose programs all live in `config/tsconfig.*.json` with no root `tsconfig.json`. `--type-aware` ran green in CI for weeks. Adding a solution-style root `tsconfig.json` took the count on an unchanged tree from 0 to 17 real findings (`restrict-template-expressions` on arrays typed `any[]`, `await-thenable`, `no-misused-promises`). Every type-aware count we had recorded before that day was taken by a checker that could not see the types, and nothing had said so.

Suggestion: when tsgolint reports `Unmatched files: N > 0`, surface it. A warning per run naming the count (and the files under `--debug=files` or similar) would have caught this on the first run. Failing the run under `--type-aware` when NO file matched any program would also be defensible, since that is never what the user asked for. I am happy to send the PR if you tell me which shape you want; I was not sure whether the warning belongs in oxlint's CLI output or in tsgolint's protocol back to oxlint.

**Comment by camc314** [comment]
Hi! Thanks for reportiing this issue and shareing the reproduction.

This is expected behavior rather than a bug: `--type-aware` enables type-aware lint rules, but does not report TypeScript errors.

In this example, `fetchThing` cannot be resolved, so `no-floating-promises` cannot identify its return value as a Promise.

Running with `--type-check` would have exposed the unresolved name:

```sh
oxlint --type-aware --type-check src
```

It is suggested to run with --type-check, to avoid typechecking and running 2x in CI as type aware linting does a lot of the same work typechecking does.

Closing as not planned - thanks!
